src/providers/glossary.provider.js

```diff
@@ -42,7 +42,7 @@ const match = (term, expression) => {
     for (let end = len; end > start; end -= 1) {
       // B = [5, 6, 7, (8), -, -] -> reduces A remaining length from right to left
       const chunk = chars.slice(start, end).join('');
-      if (chunk.toLowerCase() === term.toLowerCase()) {
+      if (chunk && term && (chunk.toLowerCase() === term.toLowerCase())) {
```

Comment:
Since `chunk` is by nature a reduction from `term`, if there's no `chunk`, then it's for sure that there's no `term`. I think an early `if (!term) return` at the beginning of the function would be a much better approach for this.